src/blocks/memory.rs

```diff
@@ -271,12 +313,50 @@ impl Memstate {
                 "SwapTotal:" => mem_state.swap_total = val,
                 "SwapFree:" => mem_state.swap_free = val,
                 "SwapCached:" => mem_state.swap_cached = val,
+                "Zswap:" => mem_state.zswap_compressed = val,
+                "Zswapped:" => mem_state.zswap_decompressed = val,
                 _ => (),
             }
 
             line.clear();
         }
 
+        // For ZRAM
+        for i in 0.. {
+            let zram_file_path = format!("/sys/block/zram{}/mm_stat", i);
+            match File::open(&zram_file_path)
+                .await
+                .error("/sys/block/zram[0-9]/mm_stat does not exist")
+            {
+                Ok(file) => {
+                    let mut zram_file = BufReader::new(file);
+                    let mut line = String::new();
```

Comment:
From https://www.kernel.org/doc/Documentation/blockdev/zram.txt:

```
File /sys/block/zram<id>/mm_stat

The stat file represents device's mm statistics. It consists of a single
line of text and contains the following stats separated by whitespace:
 orig_data_size   uncompressed size of data stored in this disk.
		  This excludes same-element-filled pages (same_pages) since
		  no memory is allocated for them.
                  Unit: bytes
 compr_data_size  compressed size of data stored in this disk
 mem_used_total   the amount of memory allocated for this disk. This
                  includes allocator fragmentation and metadata overhead,
                  allocated for this disk. So, allocator space efficiency
                  can be calculated using compr_data_size and this statistic.
                  Unit: bytes
 mem_limit        the maximum amount of memory ZRAM can use to store
                  the compressed data
 mem_used_max     the maximum amount of memory zram have consumed to
                  store the data
 same_pages       the number of same element filled pages written to this disk.
                  No memory is allocated for such pages.
 pages_compacted  the number of pages freed during compaction
 huge_pages	  the number of incompressible pages
 ```
Since there's only one line in the file then there's no need to iterate over the lines.

I'd probably also remove the error on trying to read the file, since it's possible it was hot unloaded between the open and actually reading the file

```rust
            if zram_file.read_to_string(&mut line).await.is_err() {
                continue;
            };
```